src/inventory_ui.cpp

```diff
@@ -986,11 +986,17 @@ void inventory_entry::reset_entry_cell_cache() const
 const inventory_entry::entry_cell_cache_t &inventory_entry::get_entry_cell_cache(
     inventory_selector_preset const &preset ) const
 {
+    bool has_invalid_lang_version = entry_cell_cache->lang_version !=
+                                    detail::get_current_language_version();
```

Comment:
You're dereferencing `entry_cell_cache` here without verifying that it has a value. Just move it to the `if( !entry_cell_cache ...)` below. The second `if` is redundant anyway.

Reply:
sorry andrei, I'm not 100% sure I'm following when you say "Just move it to the if( !entry_cell_cache ...) below." .  You mean moving that check at line 989 inside the first if block?  Unless I'm missing something that needs to stay outside, because there are cases in which the entry_cell_cache is valid so that if block would be skipped without that OR condition check.
Maybe I'm just misunderstanding what you meant, if it's not trouble could you show me the code you meant?

Reply:
this is what I think you meant:

> if( !entry_cell_cache ||
>         entry_cell_cache->lang_version != detail::get_current_language_version() ){
>         ...
> }
Can you confirm it's correct? Even in this case aren't we dereferencing entry_cell_cache without checking it first? Or maybe is something about the order of the checks I'm not understanding very well. Anyway, can you confirm the change is correct? : )

Reply:
`!entry_cell_cache` does the check, and if it does not contain a valid value, the or logic operator is short-circuited so the second condition will not be evaluated.